pandas/tests/indexes/test_base.py

```diff
@@ -125,88 +126,75 @@ def test_construction_list_tuples_nan(self, na_value, vtype):
         expected = MultiIndex.from_tuples(values)
         tm.assert_index_equal(result, expected)
 
-    def test_constructor_from_index_datetimetz(self):
-        idx = pd.date_range('2015-01-01 10:00', freq='D', periods=3,
-                            tz='US/Eastern')
-        result = pd.Index(idx)
-        tm.assert_index_equal(result, idx)
-        assert result.tz == idx.tz
-
-        result = pd.Index(idx.astype(object))
-        tm.assert_index_equal(result, idx)
-        assert result.tz == idx.tz
-
-    def test_constructor_from_index_timedelta(self):
-        idx = pd.timedelta_range('1 days', freq='D', periods=3)
-        result = pd.Index(idx)
-        tm.assert_index_equal(result, idx)
-
-        result = pd.Index(idx.astype(object))
-        tm.assert_index_equal(result, idx)
-
-    def test_constructor_from_index_period(self):
-        idx = pd.period_range('2015-01-01', freq='D', periods=3)
-        result = pd.Index(idx)
-        tm.assert_index_equal(result, idx)
-
-        result = pd.Index(idx.astype(object))
-        tm.assert_index_equal(result, idx)
-
-    def test_constructor_from_series_datetimetz(self):
-        idx = pd.date_range('2015-01-01 10:00', freq='D', periods=3,
-                            tz='US/Eastern')
-        result = pd.Index(pd.Series(idx))
-        tm.assert_index_equal(result, idx)
-        assert result.tz == idx.tz
+    @pytest.mark.parametrize("cast_as_obj", [True, False])
+    @pytest.mark.parametrize("idx,has_tz", [
+        (pd.date_range('2015-01-01 10:00', freq='D', periods=3,
+                       tz='US/Eastern'), True),  # datetimetz
+        (pd.timedelta_range('1 days', freq='D', periods=3), False),  # td
+        (pd.period_range('2015-01-01', freq='D', periods=3), False)  # period
+    ])
+    def test_constructor_from_index_dtlike(self, cast_as_obj, idx, has_tz):
+        if cast_as_obj:
+            result = pd.Index(idx.astype(object))
+        else:
+            result = pd.Index(idx)
 
-    def test_constructor_from_series_timedelta(self):
-        idx = pd.timedelta_range('1 days', freq='D', periods=3)
-        result = pd.Index(pd.Series(idx))
         tm.assert_index_equal(result, idx)
-
-    def test_constructor_from_series_period(self):
-        idx = pd.period_range('2015-01-01', freq='D', periods=3)
+        if has_tz:
```

Comment:
I suppose this could also leverage hasattr instead but I felt the explicitness of the parameter is more useful